autowarefoundation/autoware.universe
# feat(pid_longitudinal_controller): add diag for control state

## Description

This PR adds diagnostic for control state. It allows us to notify error when control state is emergency state.
There are some sample images.

### OK
![2022-10-18_16-59_2](https://user-images.githubusercontent.com/9547070/196372069-3141b2c9-132c-4d96-b184-e8c6f23ba38f.png)

### Translation is larger than threshold
![2022-10-18_16-59_1](https://user-images.githubusercontent.com/9547070/196372092-731d4b1a-fe97-4de9-91d7-7736b8061549.png)

### Rotation is larger than threshold
![2022-10-18_16-59](https://user-images.githubusercontent.com/9547070/196372111-4fdeba10-ad1f-44ec-a15a-ab0665773c32.png)

## Related links

### TierIV Internal Link
https://tier4.atlassian.net/browse/T4PB-20935


## Tests performed

- launch psim
- set 2d pose estimate and 2d nav goal
- launch `ros2 run rqt_robot_monitor rqt_robot_monitor`
- confirm `controller_node: control_state`
- move ego pose in some positions

## Notes for reviewers



## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
